src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -56,6 +56,7 @@ public boolean start() throws Exception {
         newTemplate.setContainers(step.getContainers());
         newTemplate.setNodeSelector(step.getNodeSelector());
         newTemplate.setServiceAccount(step.getServiceAccount());
+        newTemplate.setAnnotations(step.getAnnotations());
```

Comment:
the initial value won't be used when loading the xml config from disk, so maybe it's better if you explicitly add a `readResolve` that sets annotations to empty list if it is null

As an example
https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java#L446